projects/project6/src/main/scala/axiom_extraction/Main.scala

```diff
@@ -0,0 +1,138 @@
+package axiom_extraction
+
+
+import net.sansa_stack.rdf.spark.io._
+import org.apache.spark.sql.SparkSession
+import org.apache.jena.riot.Lang
+import org.apache.spark.rdd._
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.apache.spark.{SparkConf, SparkContext}
+import scala.collection.mutable.ListBuffer
+import org.apache.spark.sql.functions._
+import org.apache.spark.ml.fpm.FPGrowth
+
+
+object Main extends App {
+
+  var DEBUG: Boolean = false
+  
+  override def main(args: Array[String]) = {
+    val spark = SparkSession.builder
+      .appName(s"AxiomExtraction")
+      .master("local[*]")
+      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .getOrCreate()
+    val resource = "test_input.nt"
+    val path = getClass.getResource(resource).getPath()
+    val df: DataFrame = spark.read.rdf(Lang.NTRIPLES)(path)
+    Console.println("===========Input DataFrames===========")
+    df.show()
+    
+    Console.println("===========Predicates===========")
+    //selecting all distinct predicates
+    val distinctValuesDF = df.select("p").distinct

```

Comment:
use the build-in function from SANSA for getting predicates, subjects or objects by doing : 
```scala
val distinctValuesDF = df.getPredicates().distint
```
the same goes for others as well.

Reply:
It seems, that dor DataFrame and even for a RDF there is no getPredicates() method. I can extract predicates for each entrz of an RDF[Triples] with getPredicate(), but that makes the code just worse.